### ISSUE TITLE
Prefetch links that 404 should fall back to the network

### ISSUE THREAD

**Issue by sw-yx** [sveltejs/kit#3977]
### Describe the bug

i had a bug reported here: https://twitter.com/clearhunch/status/1494317555275677696?s=21

that essentially came from me using a prefetch link (`<a svelte:prefetch href=“/LIP”>`) for a url that only exists as a serverside redirect (in netlify). in a normal a href context that would work. but sveltekit is being too smart, or not smart enough, here. it checks its cache for a page called “LIP”, and not finding it, returns a 404 and end of story. i think it should deopt from clientside navigation and go actually check the network, and let the 404 handling happen there (again via sveltekit).

see what i mean? basically the only reason we do clientside nav is to make it fast, but we dont have to be fast for 404’s.

### Reproduction

the diff i made https://github.com/sw-yx/swyxdotio/commit/4f16ef636cf4e98961fa03cf63e5ad5f25a60a4b

its really obvious but it is an abstraction leak, and yes i know i could just remove prefetch but why make me check 

### System Info

```shell
n/a
```

### Severity

annoyance

**Comment by geoffrich** [comment]
This sounds related to #3935, though that issue doesn't call out prefetch specifically.

**Comment by Rich-Harris** [comment]
Just merged and released #3969 — does that solve this issue @sw-yx? 

Aside: if you implemented the server-side redirect as a `load` function instead you could avoid the roundtrip:

```svelte
<!-- src/routes/LIP.svelte -->
<script>
  export function load() {
    return {
      status: 308,
      redirect: '/learn-in-public
    };
  }
</script>
```

**Comment by sw-yx** [comment]
i think it does, will close for now as something else broke in swyxkit when i updated deps so i am debugging multiple things at once

thank you
